### PR TITLE
Joe/employe dto refractor

### PR DESCRIPTION
**Pull request by KAYEMBAJONATHAN**
Refactoring the employee Dto

All related unit tests are passing

### CODE REVIEW

File: RGO.App.Tests/Controllers/HRIS/EmployeeRoleManagerControllerUnitTests.cs [hunk 1]
```diff
@@ -105,7 +105,7 @@ public async Task AddRoleExistingRoleReturnsCreatedAtAction()
         employeeServiceMock.Setup(x => x.GetEmployee(email)).ReturnsAsync(CreateEmployee(email));
 
         roleServiceMock.Setup(x => x.CheckRole(role)).ReturnsAsync(true);
-        roleServiceMock.Setup(x => x.GetRole(role)).ReturnsAsync(new RoleDto(1, "Super Admin"));
+        roleServiceMock.Setup(x => x.GetRole(role)).ReturnsAsync(new RoleDto{ Id = 0, Description = "Super Admin" });
```
Comment (Juanro-Minne) [inline RGO.App.Tests/Controllers/HRIS/EmployeeRoleManagerControllerUnitTests.cs:108]: Just double check that you initialize the correct values everywhere
(Id was 1 you initialized it as 0)

File: RGO.Services.Tests/Handler/Charts/SalaryTypeUnitTest.cs [hunk 2]
```diff
@@ -28,7 +28,7 @@ public SalaryTypeUnitTest()
         salaryType = new SalaryType();
         employeeTypeDto = new EmployeeTypeDto(1, "Developer");
         employeeType = new EmployeeType(employeeTypeDto);
-        roleDto = new RoleDto(3, "Employee");
+        roleDto = new RoleDto{ Id = 0, Description = "Employee"};
```
Comment (Juanro-Minne) [inline RGO.Services.Tests/Handler/Charts/SalaryTypeUnitTest.cs:31]: Initial value was 3

**Review by Juanro-Minne — APPROVED** [review]

File: RGO.App.Tests/Controllers/HRIS/EmployeeRoleManagerControllerUnitTests.cs [hunk 3]
```diff
@@ -455,12 +455,13 @@ public async Task GetAllRolesReturnsOk()
 
         var roles = new List<RoleDto>
         {
-            new(1, "Super Admin"),
-            new(2, "Admin")
+            new RoleDto{ Description = "Super Admin" },
```
Comment (Juanro-Minne) [inline RGO.App.Tests/Controllers/HRIS/EmployeeRoleManagerControllerUnitTests.cs:458]:
```suggestion
            new RoleDto{ Id = 1, Description = "Super Admin" },
```

File: RGO.App.Tests/Controllers/HRIS/EmployeeRoleManagerControllerUnitTests.cs [hunk 4]
```diff
@@ -455,12 +455,13 @@ public async Task GetAllRolesReturnsOk()
...
+            new RoleDto{Id = 1, Description = "Admin"},
```
Comment (Juanro-Minne) [inline RGO.App.Tests/Controllers/HRIS/EmployeeRoleManagerControllerUnitTests.cs:459]: this id needs to be 2
```suggestion
            new RoleDto{Id = 2, Description = "Admin"},
```

**Review by Juanro-Minne — APPROVED** [review]

**Review by EstiaanBritz — APPROVED** [review]

File: RGO.Services.Tests/Services/EmployeeRoleServiceUnitTest.cs [hunk 5]
```diff
@@ -283,11 +283,11 @@ public async Task UpdateEmployeeRoleTest()
         Assert.Equivalent(employeeRoleList[1].ToDto(), result2);
         Assert.Equivalent(employeeRoleList[2].ToDto(), result3);
 
-        Assert.ThrowsAsync<Exception>(() => _employeeRoleService.UpdateEmployeeRole(new EmployeeRoleDto
+        _ = Assert.ThrowsAsync<Exception>(() => _employeeRoleService.UpdateEmployeeRole(new EmployeeRoleDto
```
Comment (tbddorothy) [inline RGO.Services.Tests/Services/EmployeeRoleServiceUnitTest.cs:286]:
```suggestion
        Assert.ThrowsAsync<Exception>(() => _employeeRoleService.UpdateEmployeeRole(new EmployeeRoleDto
```

**Review by tbddorothy — CHANGES_REQUESTED** [review]

**Review by Juanro-Minne — APPROVED** [review]

File: RGO.Services.Tests/Services/EmployeeRoleServiceUnitTest.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (KAYEMBAJONATHAN) [reply RGO.Services.Tests/Services/EmployeeRoleServiceUnitTest.cs]: @tbddorothy  fixed

File: RGO.Services.Tests/Services/EmployeeRoleServiceUnitTest.cs [hunk 6]
```diff
@@ -507,10 +507,10 @@ public async Task GetAllEmployeeOnRolesTest()
...
         {
-            new(1, "Admin"),
-            new(2, "Manager"),
-            new(3, "Employee"),
-            new(4, "Intern")
+             new RoleDto{Id = 1, Description = "Admin" },
+             new RoleDto { Id = 2, Description = "Manager" },
+             new RoleDto { Id = 3, Description = "Employee" },
+             new RoleDto { Id = 4, Description = "Intern" }
         };
```
[the remark this answers is not in the document]
Comment (KAYEMBAJONATHAN) [reply RGO.Services.Tests/Services/EmployeeRoleServiceUnitTest.cs:514]: alaright

**Review by tbddorothy — APPROVED** [review]

**Review by tbddorothy — APPROVED** [review]

**Review by Juanro-Minne — APPROVED** [review]

**Review by EstiaanBritz — APPROVED** [review]

**Review by EstiaanBritz — APPROVED** [review]

**Review by ReinoRabbit — APPROVED** [review]

**Pull request merged**